purchase_order_product_recommendation/wizards/purchase_order_recommendation.py

```diff
@@ -71,18 +79,19 @@ def _find_move_line(self, src='internal', dst='customer'):
         product_tmpls = supplierinfos.mapped('product_tmpl_id')
         products = supplierinfos.mapped('product_id')
         products |= product_tmpls.mapped('product_variant_ids')
+        domain = [
+            ('product_id', 'in', products.ids),
+            ('date', '>=', '{} 00:00:00'.format(self.date_begin)),
+            ('date', '<=', '{} 23:59:59'.format(self.date_end)),
+            ('location_id.usage', '=', src),
+            ('location_dest_id.usage', '=', dst),
+            ('state', '=', 'done'),
+        ]
+        if self.warehouse_ids:
+            domain += [('picking_id.picking_type_id.warehouse_id', 'in',
+                        self.warehouse_ids.ids)]
         found_lines = self.env['stock.move.line'].read_group(
-            [
-                ('product_id', 'in', products.ids),
-                ('date', '>=', '{} 00:00:00'.format(self.date_begin)),
-                ('date', '<=', '{} 23:59:59'.format(self.date_end)),
-                ('location_id.usage', '=', src),
-                ('location_dest_id.usage', '=', dst),
-                ('state', '=', 'done'),
-            ],
-            ['product_id', 'qty_done'],
-            ['product_id'],
-        )
+            domain, ['product_id', 'qty_done'], ['product_id'])
         # Manual ordering that circumvents ORM limitations
```

Comment:
I would like to know more about this ordering. Maybe deserve to be a tool for any read_group as an option

Reply:
The problem is that you can't limit the `read_group` operation, because you don't know totals before performing the query, so we sort according that totals and limit to the top n ones.